pkg/schedule/operator/metrics.go

```diff
@@ -82,10 +85,16 @@ var (
 
 func init() {
 	prometheus.MustRegister(operatorStepDuration)
-	prometheus.MustRegister(OperatorLimitCounter)
+	prometheus.MustRegister(operatorLimitCounter)
 	prometheus.MustRegister(OperatorExceededStoreLimitCounter)
 	prometheus.MustRegister(operatorCounter)
 	prometheus.MustRegister(operatorDuration)
 	prometheus.MustRegister(operatorSizeHist)
 	prometheus.MustRegister(storeLimitCostCounter)
 }
+
+// IncOperatorLimitCounter increases the counter of operator meeting limit.
+func IncOperatorLimitCounter(typ, name string) {
+	operatorLimitCounter.WithLabelValues(
+		strings.ReplaceAll(typ, "_", "-"), name).Inc()
```

Comment:
Is it possible not to replace it every time?

Reply:
Is it for performance reasons? In that case, we still need to cache one more xxxx-xxxx for monitoring, which brings us back to the issue.

Reply:
How about using a map to do this?

Reply:
https://github.com/tikv/pd/pull/8316/commits/aadb26d8fdc83a93e7f6b7843866f63c94118262